python/src/scipp/compat/mantid.py

```diff
@@ -513,3 +516,100 @@ def load(filename="",
     return from_mantid(data_ws,
                        load_pulse_times=load_pulse_times,
                        error_connection=error_connection)
+
+
+def load_component_info_from_instrument_file(ds, file):
+    """
+    Adds the component info labels into the dataset. The following are added:
+
+    - source_position
+    - sample_position
+    - position
+
+    :param ds: Dataset on which the component info will be added as labels.
+    :param file: File from which the IDF will be loaded.
+                 This can be anything that mantid.Load can load.
+    """
+    try:
+        import mantid.simpleapi as mantid
+    except ImportError:
+        raise ImportError(
+            "Mantid Python API was not found, please install Mantid framework "
+            "as detailed in the installation instructions (https://scipp."
+            "readthedocs.io/en/latest/getting-started/installation.html)")
+    ws = mantid.Load(file)
+
+    source_pos, sample_pos = make_component_info(ws)
+
+    ds.labels["source_position"] = source_pos
+    ds.labels["sample_position"] = sample_pos
+    ds.labels["position"] = init_pos(ws)
+
+
+def validate_dim_and_get_mantid_string(unit_dim):
+    known_units = {
+        sc.Dim.EnergyTransfer: "DelteV",
+        sc.Dim.Tof: "TOF",
+        sc.Dim.Wavelength: "Wavelength",
+        sc.Dim.Energy: "Energy",
+        sc.Dim.DSpacing: "dSpacing",
+        sc.Dim.Q: "MomentumTransfer",
+        sc.Dim.QSquared: "QSquared",
+    }
+
+    if unit_dim not in known_units.keys():
+        raise RuntimeError("Axis unit not currently supported."
+                           "Possible values are: {}, "
+                           "got '{}'. ".format([k for k in known_units.keys()],
+                                               unit_dim))
+    else:
+        return known_units[unit_dim]
+
+
+def to_mantid(x, y, e, coord_dim, ws_name):
```

Comment:
Would it be worth being explicit about which type of workspace you expect to get returned? For example, most data could be represented in a TableWorkspace, but most callers would not expect to get one.

Reply:
I don't think "most data could be represented in a TableWorkspace" is true: Yes, `Dataset` corresponds to `TableWorkspace`, but *only if it is 1D*. In all other cases this should probably just throw.